pkg/login/social/connectors/generic_oauth.go

```diff
@@ -265,6 +266,15 @@ func (s *SocialGenericOAuth) UserInfo(ctx context.Context, client *http.Client,
 			}
 		}
 
+		if len(externalOrgs) == 0 && !s.info.SkipOrgRoleSync {
+			var err error
+			externalOrgs, err = s.extractOrgs(data.rawJSON)
+			if err != nil {
+				s.log.Warn("Failed to extract orgs", "err", err)
+				return nil, err
```

Comment:
I'm not sure we want to return an error here - it's possible that the required info is not in the token (that we parse first), but is fetched through the API. I think it's enough to log a warning and move on.

Reply:
I added a test (`"Should get orgs from API when not in token"`) showing this is already the case:  no error is thrown when the attribute patch doesn't match. Error is returned only when the JMesPath is invalid.

Reply:
@IevaVasiljeva I think the `org_attribute_path` is a bit different from the rest of the settings like `groups_attribute_path`, `role_attribute_path`, etc. because if those are incorrectly configured then the user won't have the required rights/group assignment in the _default org_. On the contrary, if `org_attribute_path` is configured incorrectly then it can happen that the particular user can access _other organizations'/default org's_ data, and I assume if `org_attribute_path` is configured then `org_mapping` is/should be configured as well, so from security point of view I'd error out. 

I considered the following two options for handling the incorrect expression set for `org_attribute_path`:
1. Error out (current implementation)
    - The assumption is that if `org_attribute_path` is configured then it's configured because `org_mapping` contains a complex mapping and to be safe and secure if the code cannot evaluate unequivocally the exact mapping then it should error out.
1. Log and skip the error, but do not change the User's org role mapping (to prevent wrongfully assigning to the _default org_)
    - In this case the user's org role mapping won't be changed (if the `org_attribute_path` expression is incorrect), which could also be problematic, because it can happen that the particular user doesn't have access to the assigned orgs anymore (either because the user has been removed from the group on the IdP side or the `org_mapping` has been changed).

Reply:
You make a fair point security wise, especially when it comes to the user being able to access the default org when it might not have been intended. I would argue that if the org mapping being applied is crucial to the user, they should use `role_attribute_strict`. But as I said, I don't feel strongly about this, and you make good points in favour of erroring out. So let's stick with that :)